V4.0 WIP

Spomky opened the pull request.

Spomky committed:
Method create added (#183)

Spomky committed:
OtherUI serialization

Spomky committed:
OtherUI serialization

Spomky committed:
New tests

dependabot-preview[bot] committed:
Upgrade to GitHub-native Dependabot (#184)

Co-authored-by: dependabot-preview[bot] <27856297+dependabot-preview[bot]@users.noreply.github.com>

Spomky committed:
Deleting Sensio Extra (#185)

* Deleting Sensio Extra
* Safe for Cose fix

Spomky committed:
Update SECURITY.md

Spomky committed:
Revert

Spomky committed:
Use of PIE encoder
New ECS/Rector in the pipeline. Obsolete tools removed

Spomky committed:
Use of PIE encoder New ECS/Rector in the pipeline. Obsolete tools removed

Spomky committed:
Use of PIE encoder New ECS/Rector in the pipeline. Obsolete tools removed

Spomky committed:
New ECS/Rector in the pipeline.

Spomky committed:
New ECS/Rector in the pipeline.

TimWolla commented:
Hi @Spomky:

I'm seeing that you took the opportunity to update the code style / perform a larger refactoring. Just as a heads up: Your PIEEncoder branch is not based off the most recent v3.3. Specifically the 3 commits of mine are missing. Wanted to let you know so that you don't run into an ugly merge conflict down the road.

Spomky committed:
New ECS/Rector in the pipeline.

Spomky committed:
Use intdiv() where possible (#191)
- Replace the manual integer division in ECSignature by an explicit call to
  intdiv().
- Use intdiv() to convert from Bits to Bytes in Hmac.

Spomky committed:
Consistently use inline doc comments for @see annotations (#190)
Add the missing seconds asterisk to indicate that the comment is a doc comment.

Spomky committed:
PHP 8.1 tests

Spomky committed:
Remove obsolete import from AppleAttestationStatementSupport (#189)
`FG\ASN1\Universal\Sequence` is not referenced within this class.

Spomky committed:
Annotations removed

Spomky committed:
Dep update

Spomky committed:
Security info updated

Spomky committed:
Obsolete properties removed

Spomky committed:
Bugs fixed

Spomky committed:
Bugs fixed

asbiin commented:
@Spomky Any chance this awesome work can be restarted one day? Laravel 9 just go out and rely now on symfony 6.0. This 4.0 version would be a really good improvement for the community. Thank you!

Spomky commented:
Hi @asbiin,

I am working on the JWT stuff that need to be SF 6 compatible as well. I will take time to move this project just after.

Spomky committed:
WIP

Spomky committed:
WIP

Spomky committed:
WIP

Spomky committed:
WIP

Spomky committed:
WIP

Spomky committed:
WIP
